### ISSUE TITLE
Online viewable reports

### ISSUE THREAD

**Issue by TWiStErRob** [TWiStErRob/github-workflows#15]
TL;DR: publish reports as gh-pages.

Location:
* Separate shadow branch in the same repo
* Separate gh-pages branch in same repo
* Dedicated repo for specific repo, like `org/repo-reports`
* Dedicated repo for whole org, like `org/gha-reports`

Hosting:
 * gh-pages or main branch, doesn't matter match, as long as Pages is set up.
 * custom domain possible in Pages
 * in case of shadow-branch only specific files are visible, I don't think HTML would be viable on raw.github, but have to check

Structure:
* depending on hosting/scope, mostly for conflict avoidance.
* possible tokens in path:
  * org
  * repo
  * branch (probably not necessary)
  * sha
  * workflow ID
  * attempt #
  * module
  * report-name (lint, detekt, unit-test, android-test, etc.)
* as long as enough discriminators exist, order doesn't matter
* workflow input with template?

For each report name we might need separate processing rules, or a separate job.

Steps:
* download artifact
* clone target repo
* move artifacts to right folder
* post-process if necessary (sed paths)
* report resulting (deployed) path in workflow call output.

Making it visible:
* deployment location for each report
* update comment in PR
* create annotations on a root file (of modules), probably noisy
  * can be reduced if there's an uber-report collating all available reports for a workflow run
  * can be reduced with report browser based on paths

Implementation details:
To account for parallel CI jobs:
* either restrict parallelism, which is probably not possible per workflow call;
* or create PRs, because their merges can be serialized without conflicts.  
  This has the potential to validate the branches, but that might be an overkill. Anyway possible if necessary.
  * steps
    * push branch (with discriminator name or hash of if length restricted)
    * auto-open PR in target repo
      * enable auto-merge
      * there should be no conflicts because of distinct paths

Clean-up:
 * wipe repo on the weekend/night?
 * retention could be configured per path in pages repo
 * steps either way
   * Clone pages repo
   * figure out which folders to delete
     * `rm -rf`
     * `git filter-repo` to remove specific from history.
   * force push
